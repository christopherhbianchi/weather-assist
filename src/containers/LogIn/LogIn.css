.logInButtons{
  border: none;
  background: #5785a7;
  width: 122px;
  color: white;
  padding: 5px 15px;
  border-radius: 25px;
}
.logInButtons:hover{
  background-color: #42bcf4;
  color: white;
  transition-duration: 0.4s;
}
.selectBox{
  border: none;
  height: 20px;
  width: 190px;
  padding-left: 7px;
  border-radius: 8px;
}
.banner{
  margin: 0;
  padding: 60px 40px 40px;
  text-align: center;
  color: #e8f2f7;
  font-size: 40px;
  font-weight: 100;
}
.logInBumper{
    padding: 2px 2px;
}
.logInForm{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: #345fa5;
  width: 260px;
  border-radius: 10px;
  margin: 0 auto;
  height: 160px;
  padding: 20px;
  box-shadow: 10px 10px 30px #000000;
}
.loginPhoto{
  position: absolute;
  z-index: -1;
  top: 0px;
  background-image: url('../../assets/carsInRain.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
  width: 100%;
}
