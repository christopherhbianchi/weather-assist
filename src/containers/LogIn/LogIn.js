import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loginUserSuccess } from '../../actions/activeUserActions.js';
import  { loadUsersSuccess } from '../../actions/usersActions.js';
//import components here
const users = require('../../resources/SeedUsers.js'); //object of objects


export class LogIn extends Component {

  constructor(){
    super();
    this.state = ({
      username:'',
      password:''
    });
  }

  async componentDidMount(){
    //making sure all our users have been seeded prior to anything else
    await this.props.loadUsersSuccess(this.seedData());
  }
  seedData = () => { //seed data from hard coded data
    let usersToSeed = {};
    for(let u in users) usersToSeed[u] = users[u];
    return usersToSeed;
  }

  handleChange = event => {
    event.preventDefault();
    const {name, value} = event.target;
    this.setState({ [name]: value });
  }//handleChange

  handleSubmit = event => {
    //we need to validate users. not simply add one to the redux store. We need a way to initialize our group
    //of users on page load as well.
    event.preventDefault();
    const allUsers = this.props.users;
    const givenUsername = this.state.username;
    const givenPassword = this.state.password;

    console.log('All users from imported json keys: ' + Object.keys(users));
    console.log('line 45 All Users: ' + allUsers);
    console.log('All keys of allUsers: ' + Object.keys(allUsers));
    console.log('All Users: ' + allUsers['cbianchi']);
    console.log('allUsers[givenUsername].password: ' + allUsers[givenUsername].password);
    if(allUsers[givenUsername]){
      const targetUser = allUsers[givenUsername];
      if(targetUser.password === givenPassword){//if username exists, check password
        this.props.loginUserSuccess(this.state);
      }
    }//outer if
    else return(alert('Incorrect username and or password'));
    
  }//handleSubmit

  render(){
    return(
      <div>
        <h2>Log In</h2>
        <form>
          <input type='text' name='username' placeholder='Username' onChange={this.handleChange}/>
          <input type='text' name='password' placeholder='Password' onChange={this.handleChange}/>
          <button onClick={this.handleSubmit}>Log In</button>
      </form>
      </div>
    );//return
  }//render
}

const mapStateToProps = (state) => ({
  users: state.users //an object of objects
});
const mapDispatchToProps = (dispatch) => ({
  loginUserSuccess: user => dispatch(loginUserSuccess(user)),
  loadUsersSuccess: usersObj => dispatch(loadUsersSuccess(usersObj))
});

export default connect(mapStateToProps, mapDispatchToProps)(LogIn);
