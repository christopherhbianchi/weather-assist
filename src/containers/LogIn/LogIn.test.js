import React from 'react';
import { shallow } from 'enzyme';
import { LogIn, mapStateToProps, mapDispatchToProps } from './LogIn.js';

describe('<LogIn />', () => {
  const loadUsersSuccessMock = jest.fn();
  const renderedComponent = shallow(<LogIn
    loadUsersSuccess={ loadUsersSuccessMock }
  />);

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });//it

  it('should have a default state', () => {
    const expected = {
      username: '',
      password: ''
    };
    expect(renderedComponent.state()).toEqual(expected);
  });//it

  it('should map the store correctly', () => {
    const mockStore = { users: {} };
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  });

  it('should dispatch loginUserSuccess', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.loginUserSuccess();
    expect(mockDispatch).toHaveBeenCalled();
  });//it

  it('should dispatch loadUsersSuccess', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.loadUsersSuccess();
    expect(mockDispatch).toHaveBeenCalled();
  });//it

  it('should dispatch registerUserSuccess', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.registerUserSuccess();
    expect(mockDispatch).toHaveBeenCalled();
  });//it



});//describe
