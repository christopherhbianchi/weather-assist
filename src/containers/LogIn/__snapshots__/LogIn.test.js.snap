// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn
    loadUsersSuccess={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="banner"
>
          WEATHER ASSIST
</h2>,
        <form
          className="logInForm"
>
          <input
                    className="selectBox"
                    name="username"
                    onChange={[Function]}
                    placeholder="Username"
                    type="text"
          />
          <br />
          <input
                    className="selectBox"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
          />
          <br />
          <div
                    className="logInBumper"
          />
          <button
                    className="logInButtons"
                    onClick={[Function]}
          >
                    Log In
          </button>
          <div
                    className="logInBumper"
          />
          <button
                    className="logInButtons"
                    onClick={[Function]}
          >
                    Register
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "WEATHER ASSIST",
          "className": "banner",
        },
        "ref": null,
        "rendered": "WEATHER ASSIST",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="selectBox"
              name="username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
/>,
            <br />,
            <input
              className="selectBox"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
/>,
            <br />,
            <div
              className="logInBumper"
/>,
            <button
              className="logInButtons"
              onClick={[Function]}
>
              Log In
</button>,
            <div
              className="logInBumper"
/>,
            <button
              className="logInButtons"
              onClick={[Function]}
>
              Register
</button>,
          ],
          "className": "logInForm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "selectBox",
              "name": "username",
              "onChange": [Function],
              "placeholder": "Username",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "selectBox",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logInBumper",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "className": "logInButtons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logInBumper",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Register",
              "className": "logInButtons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Register",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="banner"
>
            WEATHER ASSIST
</h2>,
          <form
            className="logInForm"
>
            <input
                        className="selectBox"
                        name="username"
                        onChange={[Function]}
                        placeholder="Username"
                        type="text"
            />
            <br />
            <input
                        className="selectBox"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
            />
            <br />
            <div
                        className="logInBumper"
            />
            <button
                        className="logInButtons"
                        onClick={[Function]}
            >
                        Log In
            </button>
            <div
                        className="logInBumper"
            />
            <button
                        className="logInButtons"
                        onClick={[Function]}
            >
                        Register
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "WEATHER ASSIST",
            "className": "banner",
          },
          "ref": null,
          "rendered": "WEATHER ASSIST",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="selectBox"
                name="username"
                onChange={[Function]}
                placeholder="Username"
                type="text"
/>,
              <br />,
              <input
                className="selectBox"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
/>,
              <br />,
              <div
                className="logInBumper"
/>,
              <button
                className="logInButtons"
                onClick={[Function]}
>
                Log In
</button>,
              <div
                className="logInBumper"
/>,
              <button
                className="logInButtons"
                onClick={[Function]}
>
                Register
</button>,
            ],
            "className": "logInForm",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "selectBox",
                "name": "username",
                "onChange": [Function],
                "placeholder": "Username",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "selectBox",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "logInBumper",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "className": "logInButtons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log In",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "logInBumper",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Register",
                "className": "logInButtons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Register",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
